Modules/_io/_iomodule.h

```diff
@@ -34,9 +34,13 @@ extern PyType_Spec winconsoleio_spec;
  * with args=NULL, and return a new reference.
  * BUT when args=Py_True is passed, they return a borrowed reference.
  */
-extern PyObject* _PyIOBase_check_readable(PyObject *self, PyObject *args);
-extern PyObject* _PyIOBase_check_writable(PyObject *self, PyObject *args);
-extern PyObject* _PyIOBase_check_seekable(PyObject *self, PyObject *args);
+struct _io_state;  // Forward decl.
```

Comment:
Here, you can define the type: ``typedef struct _io_state _PyIO_State;`` to use ``_PyIO_State`` to use in these functions. Later, you can define the structure (members).

See ``Include/pytypedefs.h`` for the principle.

Reply:
Yeah, that looks better. Thanks, I've applied that!